### ISSUE TITLE
Unable to parse template function with return type that contains the template parameter

### ISSUE THREAD

**Issue by jasjuang** [20Tauri/DoxyDoxygen#76]
I am on doxydoxygen evolution.

Here is the minimal code to reproduce the problem

``````
class Foo
{
 public:
  template <typename T>
  std::shared_ptr<Channel<T>> MakeChannel();
};
``````

If I do alt+Q, I get a [doxy] cannot parse definition error. However, if I change the return type from T into something else like like float 

``````
class Foo
{
 public:
  template <typename T>
  std::shared_ptr<Channel<float>> MakeChannel();
};
``````

This works without a problem. Is it possible to change it so doxydoxygen is able to parse template function with return type that contains the template parameter?

**Comment by 20Tauri** [comment]
Please refer to https://github.com/20Tauri/DoxyDoxygen/issues/45,
Without a whole parser it's may be difficult to make the difference between ">>" (shift operator) and "> >".
As you can read in the previous topic, "my previous fix was not perfect"

**Comment by jasjuang** [comment]
You are right, if I simply just change the return type to T it will work, so it's the combination of ">>" and T. Look forward to the fix!

**Comment by 20Tauri** [comment]
This was fixed in 0.65.0.
